velox/exec/tests/utils/QueryAssertions.cpp

```diff
@@ -394,7 +394,27 @@ std::vector<MaterializedRow> materialize(
         row.push_back(
             rowVariantAt(dataChunk->GetValue(j, i), rowType->childAt(j)));
       } else if (isDecimalKind(typeKind)) {
-        row.push_back(duckdb::decimalVariant(dataChunk->GetValue(j, i)));
+        auto val = dataChunk->GetValue(j, i);
```

Comment:
This issue is reproducible with the standard AVG function which returns a double value instead of a decimal value. I opened an issue here: https://github.com/duckdb/duckdb/issues/6829.

I would have to modify TPC-H queries in Velox microbenchmarks to use the custom function if I went this route.

Reply:
Check if DuckDB allows built-in function override.

Reply:
I explored this idea and it does not seem to be possible at the moment.

DuckDB offers UDF API for C++ that only applies to scalar and vector functions. It does not mention anything about aggregated functions: https://duckdb.org/docs/api/cpp.html, https://github.com/duckdb/duckdb/issues/471.

Also, it does not seem possible to overwrite built-in functions. When I try running something like this: 
```c++
#include "duckdb.hpp"
#include <iostream>

using namespace duckdb;

int32_t udf_test(int32_t a) {
  return a;
}

int main(int argc, char **argv) {
   DuckDB db(nullptr);
   Connection con(db);

   con.CreateScalarFunction<int32_t, int32_t>("avg", {LogicalType::INTEGER}, LogicalType::INTEGER, &udf_test);

   auto res = con.Query("select avg(12);");
   std::cout << std::fixed << res->ToString() << std::endl;
}
```

The program crashes with 
```
libc++abi: terminating with uncaught exception of type std::length_error: vector
make: *** [run] Abort trap: 6
```

If I update the name of the function to something like `u_avg`, the function works and I get a result.

Reply:
Thanks for exploring this.

Like we discussed the reservations are about exercising this code path for all test cases. That could hide problems where the conversion from double -> decimal is not expected.

Its better to have an option for this conversion which is explicitly set in the test.